### PR TITLE
Makes ai static be removed from turfs vis_contents after ai leaves the camera chunk and cleans up cameranet code

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
yes, the two ai static overlays stayed inside a turfs vis_contents forever once they are added unless the camera chunk updates and the turf becomes visible. this means that after an ai looks over an area and the camera chunk gets created, any tiles that cant be seen by cameras have an extra two vis_contents added to them and they will most likely stay forever. sendmaps doesnt care about whether the mob can actually see anything its sending, all it cares is sending turfs in view range so any maint rooms by a high traffic hallway or a workspace gets an extra two vis_contents processed per tile for every client that sees it.

i was heavily considering nerfing or removing the ai detector traitor item, because there are two types of static that get added to the vis_contents of turfs, opaque which normal camera mobs like ais see and transparent which is only used by the ai detector traitor item. since i almost never see that traitor item being used its currently doubling the number of vis_contents added to turfs not visible by cameras. its less of a problem now though since those vis_contents are actually being removed

## Why It's Good For The Game
maptick go down for rounds that have an ai

### CODE REVIEW

**Comment by necromanceranne** [comment]
You say useless but I actually see it brought semi regularly and it is actually a fairly useful item. The AI is such a pain in the dick that knowing where it is looking is a significant boon.

But obviously if this is causing potential performance side effects, that should come first and foremost in fixing.

**Comment by Kylerace** [comment]
> 
> 
> You say useless but I actually see it brought semi regularly and it is actually a fairly useful item. The AI is such a pain in the dick that knowing where it is looking is a significant boon.
> 
> But obviously if this is causing potential performance side effects, that should come first and foremost in fixing.

ive never seen it used but i am a dirty ai main who specializes in being a pain in the dick so i might be biased. still considering it but if i do touch it ill try and make it work differently instead of removing it

**Comment by Kylerace** [comment]
Have a test merge on manuel that would get ruined if the maptick per player changed. Be back in a few days

**Comment by Kylerace** [comment]
could have just set it to draft

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm [hunk 1]
```diff
@@ -89,13 +87,11 @@ GLOBAL_DATUM_INIT(cameranet, /datum/cameranet, new)
 		var/list/remove = eye.visibleCameraChunks - visibleChunks
 		var/list/add = visibleChunks - eye.visibleCameraChunks
 
-		for(var/chunk in remove)
-			var/datum/camerachunk/c = chunk
-			c.remove(eye, FALSE)
+		for(var/datum/camerachunk/chunk in remove)
+			chunk.remove(eye, FALSE)
```
Comment (MrPerson) [inline code/modules/mob/living/silicon/ai/freelook/cameranet.dm:91]: Shouldn't this have `as anything`?

**Comment by Kylerace** [comment]
breaks done

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm (hunk 1 above)
Comment (Kylerace) [reply code/modules/mob/living/silicon/ai/freelook/cameranet.dm:91]: yes

**Comment by LemonInTheDark** [comment]
You wanna profile this on live?

**Comment by Kylerace** [comment]
> 
> 
> You wanna profile this on live?

@LemonInTheDark its not really an important change, i dont expect it to make an easily readable difference by itself except maybe in rounds where people go around breaking cameras so theres a ton of static in the hallways

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mob/living/silicon/ai/freelook/cameranet.dm:91]: 🗞️

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm [hunk 2]
```diff
@@ -1,34 +1,49 @@
-#define UPDATE_BUFFER 25 // 2.5 seconds
+#define UPDATE_BUFFER_TIME 25
```
Comment (LemonInTheDark) [inline code/modules/mob/living/silicon/ai/freelook/chunk.dm:1]: why yeet the comment? also should this be `(2.5 SECONDS)`?

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm [hunk 3]
```diff
@@ -1,34 +1,49 @@
...
+/// Add an AI eye to the chunk, then update if changed.
 /datum/camerachunk/proc/add(mob/camera/ai_eye/eye)
 	eye.visibleCameraChunks += src
 	seenby += eye
 	if(changed)
 		update()
+	else
+		var/list/vis_contents_objects = GLOB.cameranet.vis_contents_objects
+		for(var/turf/obscured_turf as anything in obscuredTurfs)
+			obscured_turf.vis_contents += vis_contents_objects
```
Comment (LemonInTheDark) [inline code/modules/mob/living/silicon/ai/freelook/chunk.dm:33]: Why the else here? I assume I'm missing something, but I'd like to make sure I understand this

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by Kylerace** [comment]
i would like to thank the 30+ contributors that made this pr possible

**Comment by LemonInTheDark** [comment]
You're just lucky it didn't request a fuck ton of reviews

**Comment by Kylerace** [comment]
> 
> 
> You're just lucky it didn't request a fuck ton of reviews

we do a little trolling

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm (hunk 3 above)
Comment (Kylerace) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm]: it was a mistake, update() would only add the static object to the vis_contents of turfs no longer visible but were last update(), now it always sets vis_contents |= vis_contents_objects so theres never a repeat

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm (hunk 2 above)
Comment (Kylerace) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm]: now defined properly

**Comment by MrStonedOne** [comment]
> its the only thing in the game that uses the transparent ai static object which gets added everywhere the visible static object is.

Explain this. I feel like this code could be optimized

**Comment by Kylerace** [comment]
dynamically removing chunks at a time is expensive, ill have to figure something else out

**Comment by Kylerace** [comment]
im leaving to get cigs be back in a bit

**Pull request closed without merging**